proxy/src/task.rs

```diff
@@ -0,0 +1,180 @@
+//! Task execution utilities.
+use futures::future::{
+    Future,
+    ExecuteError,
+    ExecuteErrorKind,
+    Executor,
+};
+use rand;
+use tokio::executor::{
+    DefaultExecutor,
+    Executor as TokioExecutor,
+    SpawnError,
+};
+
+use std::error::Error;
+use std::{fmt, io};
+
+
+/// An empty type which implements `Executor` by lazily  calling
+/// `tokio::executor::DefaultExecutor::current().execute(...)`.
+///
+/// This can be used when we would simply like to call `tokio::spawn` rather
+/// than explicitly using a particular executor, but need an `Executor` for
+/// a generic type or to pass to a function which expects one.
+///
+/// Note that this uses `DefaultExecutor` rather than `tokio::spawn`, as we
+/// would prefer for our `Executor` implementation to pass errors rather than
+/// panicking (as `tokio::spawn` does).
+#[derive(Copy, Clone, Debug, Default)]
+pub struct LazyExecutor;
```

Comment:
Makes me wonder if this should be in tokio somehow... I don't know 🤷‍♀️

Reply:
Hmm, I think it might be good to have something like this eventually?

The _implementation_ was extremely straightforward, but I didn't think of doing something like this until @seanmonstar suggested it, and it would have saved a lot of pain...maybe having it in `tokio` would make the pattern more obvious?